napari/types.py

```diff
@@ -1,6 +1,50 @@
-from typing import Callable, List, Tuple, Union, Any, Dict
+from functools import wraps
+from typing import Any, Callable, Dict, List, Protocol, Tuple, Union
+
+import numpy as np
+
+
+class ArrayLike(Protocol):
+    """This is a WOEFULLY inadqueate stub for a duck-array type.
+
+    Mostly, just a placeholder for the concept of needing an ArrayLike type.
+    Ultimately, this should come from https://github.com/napari/image-types
+    """
+
+    shape: Tuple[int, ...]
+    ndim: int
+    dtype: np.dtype
+
 
 # layer data may be: (data,) (data, meta), or (data, meta, layer_type)
-# using "Any" for the data type for now
+# using "Any" for the data type until ArrayLike is more mature.
 LayerData = Union[Tuple[Any], Tuple[Any, Dict], Tuple[Any, Dict, str]]
-ReaderFunction = Callable[[str], List[LayerData]]
+
+PathLike = Union[str, List[str]]
+ReaderFunction = Callable[[PathLike], List[LayerData]]
+
+
+def array_return_to_layerdata_return(
```

Comment:
I love everything about this except the name. =) I might call it `image_reader_to_layerdata_reader`?